lib/state/state.js

```diff
@@ -517,36 +511,72 @@ State.prototype.respond = function (response, isJson) {
 	this.response = isJson ? response : JSON.stringify(response);
 };
 
+function destroy(error, state, response, cb) {
+	// cleanup
+	state.destroy();
+
+	// return the response
+	if (cb) {
+		cb(null, response);
+	}
+}
+
+function respond(error, state, cb) {
+	const myEvents = parseEvents(error, state.myEvents);
+	state.myEvents = undefined;
+
+	destroy(error, state, {
+		response: state.response,
+		myEvents: myEvents.length ? myEvents : undefined
+	}, cb);
```

Comment:
No, that error is used to let parseEvents that we wish to retrieve events while we are in an error state.